### PR TITLE
Fix single line comment

### LINKED ISSUE #291: Single semicolon treated like code line
### Description

Single semicolon (comment) treated like code line

### Expected formatted output snippet

```autohotkey
if (expression)
    ;
    code
```

### Actual formatted output snippet

```autohotkey
if (expression)
    ;
code
```

### PR DESCRIPTION
**Pull request by kyklish**
Closes #291.

Changes proposed in this pull request:

- more strict regexp for single line  comment

Notifying @mark-wiemer

### CODE REVIEW

File: src/test/suite/providers/formatting/formattingProvider.utils.test.ts [hunk 1]
```diff
@@ -315,13 +315,17 @@ suite('FormattingProvider utils', () => {
                 rs: 'str = ""',
             },
             {
-                in: 'str = "; comment with double quote"',
```
Comment (mark-wiemer) [inline src/test/suite/providers/formatting/formattingProvider.utils.test.ts:318]: Let's keep this `in` and update the corresponding `rs`. We can add the new `'str = " ; comment"'` as well for completeness.

Same for line 322 (4 lines below this one)

File: src/providers/formattingProvider.ts [hunk 2]
```diff
@@ -279,6 +279,24 @@ export const internalFormat = (
...
+    const ahkFormatBlockCommentOn = /;\s*@AHK\+\+FormatBlockCommentOn/i;
+    /** Formatter's directive `;@AHK++FormatBlockCommentOff` */
+    const ahkFormatBlockCommentOff = /;\s*@AHK\+\+FormatBlockCommentOff/i;
+    /**
+     * Semicolon must have at least one space or tab to its left or semicolon
+     * placed at the beginning of a line.
+     *
+     * Example: `ToolTip;NotComment`, `ToolTip ;Comment`
+     *
+     * Must be in sync with comment regexp in `purify()` (formatting utility
```
Comment (mark-wiemer) [inline src/providers/formattingProvider.ts:296]: "Must be in sync" -- can we make them both an imported variable from some "constants.ts" file to enforce this?

**Review by mark-wiemer — CHANGES_REQUESTED** [review]

File: src/test/suite/providers/formatting/formattingProvider.utils.test.ts (hunk 1 above)
Comment (kyklish) [reply src/test/suite/providers/formatting/formattingProvider.utils.test.ts]: Done.

**Review by mark-wiemer — APPROVED** [review]

File: src/providers/formattingProvider.ts (hunk 2 above)
Comment (kyklish) [reply src/providers/formattingProvider.ts]: Done.

**Pull request merged**